### ISSUE TITLE
pat-inject: Unsupported different urls for multi-inject

### ISSUE THREAD

**Issue by cillianderoiste** [Patternslib/Patterns#406]
It could be useful to support multiple urls for injection.

```
<a href="full-page.html" class="pat-inject" 
     data-pat-inject="source: #sidebar; target: #sidebar; url: sidebar.html &&
                               source: #breadcrumbs; target: #breadcrumbs; url: breadcrumbs.html">click me</a>
```
In this example the default href would point to a full page which includes the sidebar and breadcrumbs (e.g. as Plone tiles) and everything else. The injections would call two separate URLs which would only render the parts of the page which we care about. This would help with performance without compromising accessibility IMO.

**Comment by cornae** [comment]
Interesting thought. Question: How would it increase performance if it results in an extra http request? (I’m thinking mobile 3/4g, latency). Or is the speed increase server side so big that that it weighs up against that?

**Comment by cillianderoiste** [comment]
Yes, it would be much less work for the server to do, and also require less work and bandwidth on the client side, but I'm really not sure what impact poor network latency would have on it.

The server wouldn't need to render parts of the page which aren't going to be injected. The client also wouldn't need to parse or download any more than it needs. It could also simplify the structure of the code and help with caching.

Maybe the injections could be executed in parallel, but I fear that would lead to syncing problems (although I don't really know how it works under the hood).

**Comment by cornae** [comment]
I think it sounds elegant. Especially since there’s a fallback to the full page which is in the HREF, which protects SEO and Accessibility.
